### DISCUSSION TITLE
piler-smtp listening on IPv6 address?

### DISCUSSION THREAD

**Discussion by Linuxmaker** [jsuto/piler#94] [Q&A] [answered]
Hello,
can piler-smtp also be configured so that it can listen on IPv6 instead of IPv4?
```
# lsof -i :25
COMMAND PID USER FD TYPE DEVICE SIZE/OFF NODE NAME
piler-smt 174306 piler 4u IPv4 17080341 0t0 TCP *:smtp (LISTEN)
```
That would be very interesting, since there are significantly more IPv6 addresses available than IPv4 addresses.

Best regards
Andreas

**Comment by jsuto** [comment]
I think it supports ipv6 already. Can you verify?

**Reply by Linuxmaker** [reply]
As I showed with lsof, piler-smpt only listens on type IPv4. You can't telnet to the IPv6 address and port 25 either. It rejects that. I'm asking because it could be that IPv6 can be supported, but only needs to be enabled. But I don't know Piler well enough for that yet, I'm just working my way into it.

**Reply by jsuto** [reply, marked as answer]
The listen address is 0.0.0.0 by default, which is an IPv4 address. Try setting ::1 or your actual ipv6 address to listen_addr parameter. It works, I've tested it.

**Comment by Linuxmaker** [comment]
Thank you very much! That was it. But it only works with either IPv6 or IPv4, but it works and that's the most important thing at the moment.
